packages/astro/src/vite-plugin-astro-server/environment.ts

```diff
@@ -0,0 +1,52 @@
+import type { AstroConfig, AstroSettings, ComponentInstance, SSRLoadedRenderer, SSRManifest } from '../@types/astro.js';
+import type { Logger } from '../core/logger/core.js';
+import type { ModuleLoader } from '../core/module-loader/index.js';
+import { Environment, loadRenderer } from '../core/render/index.js';
+import { RouteCache } from '../core/render/route-cache.js';
+import { viteID } from '../core/util.js';
+import { isServerLikeOutput } from '../prerender/utils.js';
+import { createResolve } from './resolve.js';
+import { AggregateError, CSSError, MarkdownError } from '../core/errors/index.js';
+import { enhanceViteSSRError } from '../core/errors/dev/index.js';
+
+export class DevEnvironment extends Environment {
+	constructor(
+		readonly loader: ModuleLoader,
+		readonly logger: Logger,
+		readonly manifest: SSRManifest,
+		readonly settings: AstroSettings,
+		readonly config = settings.config,
+	) {
+		const mode = 'development'
+		const resolve = createResolve(loader, settings.config.root);
+		const serverLike = isServerLikeOutput(settings.config);
+		const streaming = true;
+		const routeCache = new RouteCache(logger, mode);
+		super(logger, manifest, mode, [], resolve, serverLike, streaming, routeCache);
+	}
+
+	async preload(filePath: URL) {
```

Comment:
Docs missing

Reply:
This is not new code. It is brought in from https://github.com/withastro/astro/blob/astro@4.3.5/packages/astro/src/vite-plugin-astro-server/index.ts#L7. I don't have the context to document it.